### PR TITLE
Add requirement for securing mechanisms to have post-verification documentation.

### PR DESCRIPTION
**Pull request by msporny**
This PR attempts to address issue #1388 by requiring that securing mechanisms document what information can and cannot be used for decision making, for what period of time after verification, when additional information is required, and when care is necessary even when the returned data was properly secured.

/cc @jyasskin

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1392.html" title="Last updated on Dec 26, 2023, 3:56 PM UTC (9561353)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1392/7b22399...9561353.html" title="Last updated on Dec 26, 2023, 3:56 PM UTC (9561353)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2021,6 +2021,13 @@ <h3>Securing Mechanisms</h3>
 algorithm</a> section.
           </li>
           <li>
+MUST document post-processing considerations for data returned from verification
+processes. This includes, but is not limited to, what information can and cannot
+be used for decision making, for what period of time after verification, when
+additional information is required, and when care is necessary even when the
+returned data was properly secured.
```
Comment (dlongley) [inline index.html:2028]: I don't understand how a securing method could dictate, in any meaningful way, what could be used for decision making or not, so I've reworded this to focus on the securing mechanism describing the nature of the data being returned instead:

```suggestion
processes. This includes, but is not limited to, what information has been
secured, for what period of time after verification, when or what additional
information might be required to assist in decision making, and when
care is necessary even when the returned data was properly secured.
```

**Review by dlongley — APPROVED** [review]

**Review by selfissued — CHANGES_REQUESTED** [review]
This is weird and not actionable, as written.  Does it impinge in the ability to use verified claims for decision-making in the application, for instance?  What is it intended to enable and prevent?

Unless this can be rewritten to clearly enable actionable and useful decision-making, we're better off leaving this confusing paragraph out.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-12-19-vcwg-special#section2-5) on 2023-12-19

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.5. Add requirement for securing mechanisms to have post-verification documentation. (pr vc-data-model#1392)

_See github pull request [vc-data-model#1392](https://github.com/w3c/vc-data-model/pull/1392)._




**Brent Zundel:** "Add requirement for securing mechanisms to have post-verification documentation".  

_See github issue [vc-data-model#1388](https://github.com/w3c/vc-data-model/issues/1388)._




**Manu Sporny:** This PR is an attempt to address a concern from Jeffrey Yaskin -- he wanted the securing mechanism specs to be very clear about what is and is not acceptable from a post-processing standpoint.  
… This results in the interface we're defining -- the expectation is that there is an interface to ensure that only secured data is returned from the securing mechanism.  
… In addition, Jeffrey wanted us to say that when you return that data back, the spec still says more about it. Such as, if you use VC-JOSE-COSE, the spec should say that that no JSON-LD processing was performed.  
… A simpler example is ... "when you verify the data and you get back the secured data, you probably shouldn't sit on that data for a year and then use the data in a production setting without reverification".  
… Something that says "don't use stale data that was only checked a year ago".  
… So Jeffrey wants something like that -- this PR creates a requirement for the securing mechanism specs to be clear about those types of things.  

**Brent Zundel:** Thank you, Manu.  
… Any comments on the PR?  

**Michael Jones:** Can you explain a little more by post-processing and why Jeffrey is worried about it?  

**Manu Sporny:** Post-processing is Jeffrey ... as he's defined it ... is any processing after verification.  
… You have a secured VC in, you run the verification algorithm, you get back the protected data and then everything after that is "post-processing".  

**Michael Jones:** So using the claims would be post-processing.  
… That seems like someone we'd want to have happen.  
… I don't know what kind of negative post-processing he's concerned about.  

**Manu Sporny:** He links to two blog posts that talk about it -- and he had general uneasiness around using the claims in ways that they had not been intended to be used.  

**Michael Jones:** I'll look at it.  

**Manu Sporny:** Thanks.  

</details>

File: index.html [hunk 2]
```diff
@@ -2021,6 +2021,13 @@ <h3>Securing Mechanisms</h3>
 algorithm</a> section.
           </li>
           <li>
+MUST document post-processing considerations for data returned from verification
+processes. This includes, but is not limited to, what information has been
+secured, for what period of time after verification, when or what additional
+information might be required to assist in decision making, and when
+care is necessary even when the returned data was properly secured.
```
Comment (TallTed) [inline index.html:2028]: Business logic (a/k/a "decision making") is known to be out-of-bounds for this specification, and I think that includes the output of verification processes, so "required" and "necessary" should not be in this text. "Caution" and/or "recommendation", on the other hand, seems fair game.
```suggestion
information might be recommended to assist in decision making, and when
caution is advised even when the returned data was properly secured.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Comment by msporny** [comment]
@selfissued wrote:
> Unless this can be rewritten to clearly enable actionable and useful decision-making, we're better off leaving this confusing paragraph out.

@jyasskin, I have attempted writing some language that might address your issue #1388. @selfissued is objecting to the language and I'm at a loss of how I could rephase the language in a way that would address your concerns while simultaneously mitigating @selfissued's concerns.

Could you two try to draft some replacement language to this PR? If we are unable to find a path forward before 2nd week of January, the WG is likely to close this PR and the corresponding issue in order to move into the CR phase.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-01-03-vcwg#section1-2) on 2024-01-03

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.2. Add requirement for securing mechanisms to have post-verification documentation. (pr vc-data-model#1392)

_See github pull request [vc-data-model#1392](https://github.com/w3c/vc-data-model/pull/1392)._




**Brent Zundel:** next up 1392. a couple of requests for changes.  

**Manu Sporny:** believe Mike Jones is opposed to this PR. have asked Jeffery Yaskin if there's other language that would work. Both Mike and Jeffery need to have a discussion on appropriate text. If can't settle let's close the issue + PR.  

**Brent Zundel:** if someone wants to suggest a path forward please do, otherwise pending close.  
… marking as pending close. hope within the 7 day window those opposed can work together and find a path forward.  

</details>

**Comment by jyasskin** [comment]
Closing this is fine with me. It doesn't address the problem I was concerned about in #1388. Sorry for the confusion that led to extra drafting effort.

**Review by David-Chadwick — APPROVED** [review]

**Pull request closed without merging**